appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/admin/SetMPHealthCheckConfiguration.java

```diff
@@ -107,13 +114,27 @@
     @Inject
     UnprocessedConfigListener unprocessedListener;
 
+    @Param(optional = true, alias = "securityenabled")
+    private Boolean securityEnabled;
+
+    @Inject
+    private CommandRunner commandRunner;
+
+    @Inject
+    private Domain domain;
+
     @Override
     public void execute(AdminCommandContext context) {
         ActionReport actionReport = context.getActionReport();
-
+        Subject subject = context.getSubject();
         Config targetConfig = targetUtil.getConfig(target);
         MetricsHealthCheckConfiguration config = targetConfig.getExtensionByType(MetricsHealthCheckConfiguration.class);
 
+        // Create the default user if it doesn't exist
+        if (!defaultMicroprofileUserExists(actionReport.addSubActionsReport(), subject)) {
+            createDefaultMicroprofileUser(actionReport.addSubActionsReport(), subject);
```

Comment:
Using `addSubActionsReport` twice instead of once sounds wrong to me. If it actually adds (modifies state) I believe this should only happen once.

Reply:
Sub ActionReport new instance is required for both function `defaultMicroprofileUserExists` and `createDefaultMicroprofileUser` to invoke `list-file-users` and `create-file-user` commands